main.go

```diff
@@ -307,20 +311,20 @@ func postBlocks(client SlackClient, postOpts *PostOptions, blocks slack.Blocks)
 	return output, nil
 }
 
-func postFile(client SlackClient, postOpts *PostOptions, fileReader io.Reader, filename, comment string) (*CliOutput, error) {
+func postFile(client SlackClient, postOpts *PostOptions, fileReader io.Reader, filename, comment string, size int) (*CliOutput, error) {
 	postTime := time.Now()
 	if filename == "" {
 		filename = fmt.Sprintf("%s.txt", postTime.Format("20060102_150405.999999"))
 	}
-	fups := slack.FileUploadParameters{
+	fups := slack.UploadFileV2Parameters{
 		Filename:        filename,
+		FileSize:        size,
```

Comment:
https://api.slack.com/methods/files.getUploadURLExternal requires a length of the file.